src/Session.cpp

```diff
@@ -446,6 +454,11 @@ bool CSession::InitializeDRM(bool addDefaultKID /* = false */)
           drmOptionalKeyParam = licenseData;
         }
       }
+      else if (licenseType == "org.w3.clearkey")
+      {
+        // URL for license server is set to PSSH, use this as is
+        initData = sessionPsshset.pssh_;
```

Comment:
this behaviour looks weird,
can you give me a sample stream where the license url is set on pssh?

Reply:
i think i have understood that this is due to dash laurl?
imo we should avoid this mix, then decouple licenseurl from pssh,
a possibility is add licenseurl to PSSHSet and add new argument licenseurl to CreateSingleSampleDecrypter

Reply:
Yes it's from laurl. I've (ab)used the pssh to get this through to the decrypter.
I remember us discussing this last year sometime, I can't remember the outcome back then but I agree it should be clearer what is happening otherwise it becomes tangled up too much later on.